### ISSUE TITLE
NuGet package?

### ISSUE THREAD

**Issue by geraldcombs** [vslavik/winsparkle#15]
Would it be possible to add a [NuGet](http://www.nuget.org) package for WinSparkle?

**Comment by spoiledtechie** [comment]
I would love to see the standard WinSparkle.dll nuget package.

**Comment by vslavik** [comment]
> I would love to see the standard WinSparkle.dll nuget package.

No kidding. The way to achieve that is by submitting a PR, not pointless "me too" comments...

**Comment by MauroAllende** [comment]
I added the the file named WinSparkle.dll in a folder within a WPF project and when I tried to add the reference to the dll I am getting the following error:

A reference to "[Path]\WinSparkle.dll" could not be added. Please make sure that the file is accessible and that it is a valid assembly or COM component.

Do you know which could be the cause of this issue?

Thanks,
Mauro

**Comment by junian** [comment]
Hi,

Great work by the way. I just got a job to implement auto update and so far this is the best tool! Unfortunately, there is no good .NET wrapper on NuGet so I decided to create the package myself.

In case some of you still need it, you can install it via NuGet from this link: https://www.nuget.org/packages/WinSparkle.DotNet/

I also wrap the API using C# code convention, so to use it you can type to something like this:

```
var _winSparkle = WinSparkle.Current;
_winSparkle.SetErrorCallback(() => { MessageBox.Show("Error Callback"); });
_winSparkle.SetCanShutdownCallback(() => { MessageBox.Show("Can Shutdown Callback"); });
_winSparkle.SetDidFindUpdateCallback(() => { MessageBox.Show("Did Find Update Callback"); });
_winSparkle.SetDidNotFindUpdateCallback(() => { MessageBox.Show("Did Not Find Update Callback"); });
_winSparkle.SetShutdownRequestCallback(() => { MessageBox.Show("Shutdown Request Callback"); });
_winSparkle.SetUpdateCancelledCallback(() => { MessageBox.Show("Update Cancelled Callback"); });

_winSparkle.SetAppcastUrl("https://yoururl.com/file.rss");
_winSparkle.Init();
_winSparkle.CheckUpdateWithUI();
```

Thanks,


Junian

**Comment by vslavik** [comment]
> Unfortunately, there is no good .NET wrapper on NuGet so I decided to create the package myself.

And instead of being a good Open Source citizen and submitting a PR with necessary changes, you went rogue and appropriated this project's name without contributing build or other changes. Not very cool.

Can you please at least rename your NuGet package to make it obvious it is not official? Such as  by including "unofficial" in its name? Thanks.

**Comment by junian** [comment]
Sorry about that, I don't have any intention to appropriate this project's name or ownership.

As your suggestion, I put "unofficial" on its title and description. Thanks.

**Comment by vslavik** [comment]
Just-released 0.8.0 has a NuGet package (C++/native only, not yet for C#).
